### ISSUE TITLE
Erroneously flagging literal case inside JSX in jsx-only mode

### ISSUE THREAD

**Issue by glitchassassin** [edvardchen/eslint-plugin-i18next#86]
In JSX-only mode, this raises a `disallow literal string` error for the line `case 'abc':`

```
<div>{[].map((a) => {
  switch (a) {
    case 'abc':
      return null;
    default:
      return null;
  }
})}</div>
```

Expected behavior: This nested JS expression should be evaluated as JS

**Comment by glitchassassin** [comment]
this does not appear to be an issue for `all` or `jsx-text-only` mode
